packages/react-devtools-shared/src/utils.js

```diff
@@ -440,12 +441,23 @@ export function getDataType(data: Object): DataType {
         return 'regexp';
       } else if (Object.prototype.toString.call(data) === '[object Date]') {
         return 'date';
+      } else if (
+        Object.prototype.toString.call(data) === '[object HTMLAllCollection]'
+      ) {
+        return 'html_all_collection';
```

Comment:
Well, old **Internet Explorer** would hit this branch, since `document.all` used to be a non‑standard **IE** extension back in the **Internet Explorer** vs. **Netscape** era and websites supporting both would do:

```js
if (document.all) {
	// IE code
} else {
	// Netscape code
}
```

But websites supporting only **IE** would just use `document.all` without checking for it.

---

To support both, **WHATWG** deemed it necessary to make `document.all` behave as `undefined`, while also being an `object`.